src/core/common/message.cpp

```diff
@@ -110,9 +127,13 @@ ThreadError MessagePool::FreeBuffers(Buffer *aBuffer)
     while (aBuffer != NULL)
     {
         tmpBuffer = aBuffer->GetNextBuffer();
+#if OPENTHREAD_CONFIG_PLATFORM_MESSAGE_MANAGEMENT
+        otPlatMessagePoolFree(static_cast<struct BufferHeader *>(aBuffer));
+#else // OPENTHREAD_CONFIG_PLATFORM_MESSAGE_MANAGEMENT
         aBuffer->SetNextBuffer(mFreeBuffers);
         mFreeBuffers = aBuffer;
         mNumFreeBuffers++;
```

Comment:
Because you are splitting variable and logic, I have no way to review or understand either piece of code when presented by itself. I can't tell that your code that implements this interface is going to be changing these variables under the covers. It just seems like really confusing design. If you are managing the messages, you should own everything (even the main buffer memory in my opinion). Just my opinion.